test/CMakeLists.txt

```diff
@@ -43,24 +43,22 @@ function(make_test src target std)
     target_link_libraries(${target} PRIVATE magic_enum::magic_enum)
     set_target_properties(${target} PROPERTIES CXX_EXTENSIONS OFF)
     if(std)
-        if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-            target_compile_options(${target} PRIVATE /std:${std})
-        else()
-            target_compile_options(${target} PRIVATE -std=${std})
-        endif()
+        target_compile_features(${target} PRIVATE cxx_std_${std})
```

Comment:
This was the reason for the double defined c++ std!

Better to use  `target_compile_features()!`